internal/relay/trusted_headers.go

```diff
@@ -10,4 +11,5 @@ import (
 type TrustedHeaderFetcher interface {
 	// Fetch returns two trusted Headers for height and height+1 packed into *codectypes.Any value
 	Fetch(ctx context.Context, height uint64) (header *codectypes.Any, nextHeader *codectypes.Any, err error)
+	FetchBest(ctx context.Context, height uint64) (exported.Header, error)
```

Comment:
I don't like the naming here: both methods actually returns _best_ headers but only one method is called `FetchBest`.
I think it's quite misleading.

Also one method is used by TX queries only, and the other one - by KV.